### PR TITLE
docs: overview for cli reference section

### PR DESCRIPTION
**Pull request by jbogarthyde**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[x] Documentation content changes
[ ] angular.io application / infrastructure changes
[ ] Other... Please describe:
```

## What is the current behavior?

Issue Number: N/A

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

## Other information
To be added to the beginning of the new CLI Reference section

### CODE REVIEW

**Comment by jenniferfell** [comment]
Waiting for https://github.com/angular/angular/pull/25363

**Review by Splaktar — CHANGES_REQUESTED** [review]
I'm glad to see this documentation coming to Angular.io! Thank you for your hard work on this 😄

File: aio/content/cli/index.md [hunk 1]
```diff
@@ -1,83 +1,201 @@
...
-Invoke the tool on the command line through the ng executable. Online help is available on the command line:
+Invoke the tool on the command line through the `ng` executable. 
+Online help is available on the command line. 
+Enter the following to list commands or options for a given command, with a short description.
 
 ```
-> ng help               Lists commands with short descriptions
-> ng <command> --help   Lists options for a command.
+> ng help
+> ng <command> --help`
```
[the remark this answers is not in the document]
Comment (jbogarthyde) [reply aio/content/cli/index.md]: Changing all of these to use <code-example format="." language="bash">

File: aio/content/cli/index.md [hunk 2]
```diff
@@ -1,83 +1,201 @@
...
+<div class="alert-is-helpful>
 
-A newly generated app project contains the source files for a root module, with a root component and template, which you can edit directly, or add to and modify using CLI commands. Use the generate command to add new files for additional components and services, and code for new pipes, directives, and so on.
+Angular 6 introduced the [workspace](guide/glossary#workspace) directory structure for Angular [projects](guide/glossary#project).  A project can be an *app* or a *library*, and a workspace can contain multiple apps, as well as libraries that can be used in any of the apps.
+Some commands (such as `build`) must be executed from within a workspace folder, and others (such as `new`) must be executed from outside any workspace.
```
[the remark this answers is not in the document]
Comment (jbogarthyde) [reply aio/content/cli/index.md]: Yes, but I don't know how to do that yet. 
@petebacondarwin ?

[the remark this answers is not in the document]
Comment (petebacondarwin) [reply aio/content/cli/index.md]: How about this?

```
[`build`](cli/build)
```

File: aio/content/cli/index.md [hunk 3]
```diff
@@ -1,83 +1,201 @@
...
+`--myOptionName` is equivalent to `--my-option-name`.
+
+### Boolean and enumerated options
+
+Boolean options have two forms: `--thisOption` sets the flag, `--noThisOption` clears it.  
+If neither option is supplied, the flag remains in its default state, as listed in the reference documentation.
+
+Allowed values and defaults are given with each enumerated option description. For example:
+
+ `--optionNameOrAlias=allowedValue1|allowedValue2|allowedValue3` : Default is `allowedValue1`.
```
[the remark this answers is not in the document]
Comment (jbogarthyde) [reply aio/content/cli/index.md]: @petebacondarwin Do you want to make this boilerplate on every page? Or perhaps repeat the default in the description, so that the bold is just a quick-reference cue?

File: aio/content/cli/index.md [hunk 4]
```diff
@@ -1,83 +1,201 @@
...
+
+### Relative paths
+
+Options that specify files can be given as absolute paths, or as paths relative to the current working directory, which is generally either the workspace or project root.
+
+### Schematics
+
+The `generate` and `add` commands take as an argument the artifact or library to be generated or added to the current project. 
+In addition to any general options, each artifact or library defines its own options in a *schematic*. 
+Schematic options are listed in a separate reference document, but are supplied to the command in the same format as immediate command options.
```
[the remark this answers is not in the document]
Comment (jbogarthyde) [reply aio/content/cli/index.md]: They are not listed separately now -- they are simply listed for each artifact, so it should be clear.

File: aio/content/cli/index.md [hunk 5]
```diff
@@ -1,83 +1,201 @@
...
+| build (b)       | Compiles an Angular app into an output directory named `dist/` at the given output path. |
+| config (c)      | Sets or retrieves Angular CLI configuration values. |
+| doc (d)         | Opens the official Angular API documentation for a given keyword on angular.io. |
+| e2e (e)         | Builds and serves an Angular app, and runs end-to-end tests using Protractor. |
+| eject           | *Temporarily disabled.* Ejects your app (from CLI build process) and outputs the proper webpack configuration and scripts for external build. |
+| generate (g)    | Creates and initializes a code skeleton for an Angular project or part of a project, based on a given schematic. |
+| help (h)        | Outputs command documentation in the console. |
+| lint (l)        | Runs the TSLint tool on Angular app code in the given project folder. |
+| new (n)         | Creates a new Angular workspace directory and initializes it with a generic app skeleton and the corresponding end-to-end test app. |
+| run (r)         | Runs unit tests in current project. |
```
[the remark this answers is not in the document]
Comment (jbogarthyde) [reply aio/content/cli/index.md]: We haven't figured out how to talk about Architect - it's not entirely public, but it does show up by name in the config file. I don't know what it means to "run an Architect target" - suggestions welcome.

File: aio/content/cli/index.md [hunk 6]
```diff
@@ -1,83 +1,201 @@
...
+
+Options that specify files can be given as absolute paths, or as paths relative to the current working directory, which is generally either the workspace or project root.
+
+### Schematics
+
+The `generate` and `add` commands take as an argument the artifact or library to be generated or added to the current project. 
+In addition to any general options, each artifact or library defines its own options in a *schematic*. 
+Schematic options are listed in a separate reference document, but are supplied to the command in the same format as immediate command options. 
+
+## Command overview
```
[the remark this answers is not in the document]
Comment (jbogarthyde) [reply aio/content/cli/index.md]: oops. I see that @petebacondarwin is generating the table from the descriptions in the source. Since I haven't been able to edit those yet, these edits won't get in for a while. 
I want to keep your corrections - how about if I leave this table here until I can edit the source for the generated table.

File: aio/content/cli/index.md [hunk 7]
```diff
@@ -1,83 +1,201 @@
...
+| ARTIFACT (ALIAS) | DESCRIPTION |
+| :--------------  | :----------------------------------------------|
+| application (a)  | Generate a new, generic app definition in the given project. |
+| class (cl)       | Generate a new, generic class definition in the given project. |
+| component (c)    | Generate a new, generic component definition in the given project. |
+| directive (d)    | Generate a new, generic directive definition in the given project. |
+| enum (e)         | Generate a new, generic enum definition in the given project. |
+| guard (g)        | Generate a new, generic route guard definition in the given project. |
+| interface (i)    | Generate a new, generic interface definition in the given project. |
+| library (l)      | Generate a new, generic library definition in the given project. |
```
[the remark this answers is not in the document]
Comment (jbogarthyde) [reply aio/content/cli/index.md]: Same goes for app

File: aio/content/cli/index.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (jbogarthyde) [reply aio/content/cli/index.md]: If this table is being replaced by your generated table, this shouldn't be an issue.

File: aio/content/cli/index.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (Splaktar) [reply aio/content/cli/index.md]: Looking forward to seeing how this renders after this change.

**Comment by Splaktar** [comment]
This PR needs to be rebased to pick up build fix from master. Then mary-poppins will be able to deploy your latest changes for verification.

**Comment by jbogarthyde** [comment]
Looks like the preview still failed.

File: aio/content/cli/index.md [hunk 8]
```diff
@@ -1,83 +1,209 @@
...
+Enter the following to list commands or options for a given command (such as [generate](cli/generate)) with a short description.
 
-```
-> ng help               Lists commands with short descriptions
-> ng <command> --help   Lists options for a command.
-```
+<code-example format="." language="bash">
+ng help
+ng generate --help`
+</code-example
```
Comment (Splaktar) [inline aio/content/cli/index.md]: This is throwing an error in the build:
```
Unmatched plain HTML block tag <code-example format="." language="bash">
    at Object.exports.prettifyError (/home/travis/build/angular/angular/aio/node_modules/nunjucks/src/lib.js:34:15)
```
There is a missing `>` at the end of `</code-example`

Comment (jbogarthyde) [reply aio/content/cli/index.md]: thanks

**Comment by Splaktar** [comment]
These TravisCI failures look like flakes (related to network connectivity and https://github.com/yarnpkg/yarn/issues/2405):
```
scripts/ci/install.sh install-yarn
curl -o- -L https://yarnpkg.com/install.sh
bash: line 1: syntax error near unexpected token `newline'
bash: line 1: `<HTML>'
```
If someone can restart the TravisCI build, it looks like it may resolve itself. I don't have that permission for angular/angular.

File: aio/content/cli/index.md [hunk 9]
```diff
@@ -1,83 +1,209 @@
...
+Online help is available on the command line. 
+Enter the following to list commands or options for a given command (such as [generate](cli/generate)) with a short description.
 
-```
-> ng help               Lists commands with short descriptions
-> ng <command> --help   Lists options for a command.
-```
+<code-example format="." language="bash">
+ng help
+ng generate --help`
```
Comment (Splaktar) [inline aio/content/cli/index.md]: This trailing bactick should be removed.

File: aio/content/cli/index.md [hunk 10]
```diff
@@ -1,83 +1,209 @@
...
+The [ng new](cli/new) command creates an *Angular workspace* folder and generates a new app skeleton. 
 
-Some commands (such as build) must be executed from within a workspace folder, and others (such as new) must be executed from outside any workspace. This requirement is called out in the description of each command where it applies.The `new` command creates a [workspace](guide/glossary#workspace) to contain [projects](guide/glossary#project). A project can be an app or a library, and a workspace can contain multiple apps and libraries.
+<div class="alert-is-helpful>
 
-A newly generated app project contains the source files for a root module, with a root component and template, which you can edit directly, or add to and modify using CLI commands. Use the generate command to add new files for additional components and services, and code for new pipes, directives, and so on.
+Angular 6 introduced the [workspace](guide/glossary#workspace) directory structure for Angular [projects](guide/glossary#project).
```
Comment (Splaktar) [inline aio/content/cli/index.md]: This line is still not being parsed properly. It parses to "workspace directory structure for Angular projects." where the "workspace" link doesn't work and the "projects" like does work. The "Angular 6 introduced the" is missing in the docs output.

@petebacondarwin the syntax here looks fine to me. Do you have any ideas?

File: aio/content/cli/index.md [hunk 11]
```diff
@@ -1,83 +1,209 @@
...
+The top level of the workspace contains global configuration files.
+Project-level configuration files allow you to customize individual projects. 
+See more about the [Workspace file structure](guide/file-structure). 
 
-All commands and some options have aliases, as listed in the descriptions. Option names are prefixed with a double dash (--), but arguments and option aliases are not.
+The CLI configuration file, `angular.json`, is created at the top level of the workspace. 
+This is where you can set global CLI defaults for the entire workspace, and specify configurations to use when the CLI builds a project for different targets.
 
-Typically, the name of a generated artifact can be given as an argument to the command or specified with the --name option. Most commands have additional options.
+The `[ng config](cli/config) command lets you set and retrieve configuration values from the command line, or you can edit the `angular.json` file directly.
```
Comment (Splaktar) [inline aio/content/cli/index.md]: The backtick in front of `[ng config](cli/config)` should be removed so that this can be parsed properly.

File: aio/content/cli/index.md [hunk 12]
```diff
@@ -1,83 +1,209 @@
...
+| run (r)         | Runs an Architect target in the current project. |
+| serve (s)       | Builds, and runs an Angular app in a local browser in development mode. Rebuilds and refreshes when any file changes. |
+| test (t)        | Builds and tests an Angular project. |
+| update (u)      | Updates the current app with the latest version of an npm package (if supported by the given package). |
+| version (v)     | Reports the version of  the CLI, its dependencies, and environment information to the console. |
+| xi18n (x)       | Extracts i18n messages from an app's templates to a file, in preparation for translation. |
+
+## Schematic overview
+
+When you use the `generate` command to create an app or part of an app, the available options are determined by the schematic for that artifact.
```
Comment (Splaktar) [inline aio/content/cli/index.md]: It would be nice to link to the given command docs on this line and the next two like above.

**Review by Splaktar — CHANGES_REQUESTED** [review]
This is looking really good! I just had a few more syntax issues to point out that caused some of the parsing the break. Oh and one minor suggestion.

File: aio/content/cli/index.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (filipesilva) [reply aio/content/cli/index.md]: `Runs an Architect target in the current project.` is a good description of this command. We wanted to have a  stable API and more information about Architect by now though...

File: aio/content/cli/index.md [hunk 13]
```diff
@@ -1,83 +1,209 @@
...
 In your browser, open http://localhost:4200/ to see the new app run.
 
-### Workspaces and project files
+## Workspaces and project files
 
-Angular 6 introduced the workspace directory structure for Angular apps. A workspace defines a project. A project can contain multiple apps, as well as libraries that can be used in any of the apps.
+The [ng new](cli/new) command creates an *Angular workspace* folder and generates a new app skeleton. 
 
-Some commands (such as build) must be executed from within a workspace folder, and others (such as new) must be executed from outside any workspace. This requirement is called out in the description of each command where it applies.The `new` command creates a [workspace](guide/glossary#workspace) to contain [projects](guide/glossary#project). A project can be an app or a library, and a workspace can contain multiple apps and libraries.
+<div class="alert-is-helpful>
```
Comment (petebacondarwin) [inline aio/content/cli/index.md]: Missing closing double quote.

**Review by Splaktar — APPROVED** [review]
LGTM!

File: aio/content/cli/index.md [hunk 14]
```diff
@@ -1,83 +1,171 @@
...
-The CLI config command lets you set and retrieve configuration values from the command line, or you can edit the angular.json file directly.
+The `[ng config](cli/config) command lets you set and retrieve configuration values from the command line, or you can edit the `angular.json` file directly.
 
-* See the complete schema for angular.json.
-* Learn more about configuration options for Angular (link to new guide?)
+* See the [complete schema](https://github.com/angular/angular-cli/wiki/angular-workspace) for `angular.json`. 
+* Learn more about *configuration options for Angular(links to new guide or topics TBD)*.
 
-### Command options and arguments
+### Configure auto-completion in a command shell
```
Comment (IgorMinar) [inline aio/content/cli/index.md]: I don't think we support autocompletion in the current version of the cli. please remove the whole section.

@clydin can you confirm?

File: aio/content/cli/index.md [hunk 15]
```diff
@@ -1,83 +1,171 @@
...
+
+
+| COMMAND (ALIAS) | DESCRIPTION |
+| :-------------- | :----------------------------------------------|
+| add (a)         | Adds support for an Angular package or library to a project. |
+| build (b)       | Compiles an Angular project into a named subfolder within the workspace's` dist/` folder. |
+| config (c)      | Sets or retrieves Angular CLI configuration values. |
+| doc (d)         | Opens the official Angular API documentation for a given keyword on angular.io. |
+| e2e (e)         | Builds and serves an Angular app, and runs end-to-end tests using Protractor. |
+| eject           | *Temporarily disabled.* Ejects your app (from CLI build process) and outputs the proper webpack configuration and scripts for external build. |
```
Comment (IgorMinar) [inline aio/content/cli/index.md]: if it's disabled we should not document it at all

File: aio/content/cli/index.md [hunk 16]
```diff
@@ -1,83 +1,171 @@
 <h1 class="no-toc">CLI Command Reference</h1>
 
-The Angular CLI is a command-line tool that you use to initialize, develop, scaffold, and maintain Angular applications.
+The Angular Command-line Interface (CLI) is a command-line tool that you use to initialize, develop, scaffold, and maintain Angular applications. You can use the tool directly in a command shell, or indirectly through an interactive UI such as [Angular Console](https://angularconsole.com).
```
Comment (IgorMinar) [inline aio/content/cli/index.md]: we never refer to the tool as command line interface. could we reword this as "Angular CLI is a command line interface tool ..." ?

**Review by IgorMinar — APPROVED** [review]
the rest looks great! thanks

File: aio/content/cli/index.md (hunk 14 above)
Comment (clydin) [reply aio/content/cli/index.md]: Correct.  It was removed in 6.0.

File: aio/content/cli/index.md (hunk 15 above)
Comment (clydin) [reply aio/content/cli/index.md]: It's more than temporarily disabled now. The following is the output if the command is run in 7.0:
```
      The 'eject' command has been disabled and will be removed completely in 8.0.
      The new configuration format provides increased flexibility to modify the
      configuration of your workspace without ejecting.

      There are several projects that can be used in conjuction with the new
      configuration format that provide the benefits of ejecting without the maintenance
      overhead.  One such project is ngx-build-plus found here:
      https://github.com/manfredsteyer/ngx-build-plus
```

File: aio/content/cli/index.md [hunk 17]
```diff
@@ -1,83 +1,171 @@
...
 
-## Getting Started
+## Installing Angular CLI
 
-### Installing Angular CLI
+Major versions of Angular CLI follow the supported major version of Angular, but minor versions can be released separately.
 
-The current version of Angular CLI is 6.x.
+In the current version, CLI 6.x, both the CLI and the projects that you generate with the tool have dependencies that require Node 8.9 or higher, together with NPM 5.5.1 or higher.
```
[the remark this answers is not in the document]
Comment (jenniferfell) [reply aio/content/cli/index.md]: Absolutely agree. I've been moving pre-reqs into a single area of Getting Started.

**Comment by jbogarthyde** [comment]
Failing Travis for unrelated reasons
